app/src/main/java/com/flowfoundation/wallet/manager/cadence/CadenceApiManager.kt

```diff
@@ -77,6 +122,48 @@ object CadenceApiManager {
         }
     }
 
+    private fun verifySignature(signature: String, data: ByteArray): Boolean {
+        try {
+            val messageDigest = MessageDigest.getInstance("SHA-256")
+            val hashedData = messageDigest.digest(data)
+
+            val pubKeyBytes = BuildConfig.X_SIGNATURE_KEY.decodeHex().toByteArray()
+
+            val ecSpec = ECNamedCurveTable.getParameterSpec("secp256r1")
+
+            val pubKeyPoint = ecSpec.curve.decodePoint(pubKeyBytes)
+
+            val pubKeyParams = ECPublicKeyParameters(pubKeyPoint,
+                ECDomainParameters(ecSpec.curve, ecSpec.g, ecSpec.n, ecSpec.h)
+            )
+
+            val rawSignatureBytes = signature.decodeHex().toByteArray()
+
+            val r: BigInteger
+            val s: BigInteger
+
+            if (rawSignatureBytes.size == 64) {
+                r = BigInteger(1, rawSignatureBytes.copyOfRange(0, 32))
+                s = BigInteger(1, rawSignatureBytes.copyOfRange(32, 64))
+            } else {
+                val asn1InputStream = ASN1InputStream(ByteArrayInputStream(rawSignatureBytes))
+                val sequence = asn1InputStream.readObject() as DERSequence
+                r = (sequence.getObjectAt(0) as ASN1Integer).value
```

Comment:
Let's use WalletCore method to verify, which is simpler
https://github.com/onflow/FRW-web-next/issues/285#issuecomment-2790043984